### PR TITLE
Refactored api.register

### LINKED ISSUE #213: Refactor api methods using User-interactive auth
Any methods of `MatrixHttpApi` using User-Interactive Authentication should have any optional or required parameters outside of the `auth` object as parameters (kwarg or otherwise) to their method-call. The auth object should also be passed explicitly as an argument.

As much as possible try to make the api for all these methods similar. See also methods introduced in #209.

### PR DESCRIPTION
**Pull request by abhishekkumar2718**
api.register now uses auth_body, kind and **kwargs. 
Closes #213

### CODE REVIEW

**Comment by abhishekkumar2718** [comment]
Signed-off-by: Abhishek Kumar abhishekkumar2718@gmail.com

File: matrix_client/api.py [hunk 1]
```diff
@@ -107,34 +107,21 @@ def sync(self, since=None, timeout_ms=30000, filter=None,
     def validate_certificate(self, valid):
         self.validate_cert = valid
 
-    def register(self, content=None, kind='user'):
+    def register(self, auth_body, kind, **kwargs):
```
Comment (non-Jedi) [inline matrix_client/api.py]: Let's explicitly list all the possible kwargs here. `bind_email`, `username`, `password` `device_id`, and `initial_device_display_name`.

Comment (non-Jedi) [reply matrix_client/api.py]: Not sure if it was clear before. I want the following listed in the function signature; not just the docstring. :)

Comment (abhishekkumar2718) [reply matrix_client/api.py]: Wouldn't that defeat the point of using kwargs ?
I think you mean a function signature like `register(self, auth_body, bind_email=None, ...)`

Comment (non-Jedi) [reply matrix_client/api.py]: Yep. I think this method would be better without kwargs with each possible top-level key listed explicitly. Just like you showed, yes.

**Comment by non-Jedi** [comment]
LGTM. If you can squash down to one commit, I'll go ahead and merge. Thanks!

File: matrix_client/api.py [hunk 2]
```diff
@@ -166,7 +168,7 @@ def login(self, login_type, **kwargs):
 
         Args:
             login_type (str): The value for the 'type' key.
-            **kwargs: Additional key/values to add to the JSON submitted.
+            **kwargs: Additional key/values to add to JSON submitted.
```
Comment (Zil0) [inline matrix_client/api.py]: Please remove this change.

**Comment by Zil0** [comment]
Merged, thanks for the time you spent on this :)

**Comment by abhishekkumar2718** [comment]
It was my pleasure! Onto the next open issue. :)

**Pull request merged**